src/main/java/org/interledger/codecs/CodecContext.java

```diff
@@ -97,36 +104,68 @@ public InterledgerPacket read(final InputStream inputStream) throws IOException
    * Helper method that accepts an {@link InputStream} and a type hint, and then decodes the input
    * to the appropriate response payload.
    *
-   * @param type An instance of {@link Class} that indicates the type that should be decoded.
+   * @param type        An instance of {@link Class} that indicates the type that should be
+   *                    decoded.
    * @param inputStream An instance of {@link InputStream} that contains bytes in a certain
-   *     encoding.
-   * @param <T> The type of object to return, based upon the supplied type of {@code type}.
+   *                    encoding.
+   * @param <T>         The type of object to return, based upon the supplied type of {@code type}.
+   *
    * @return An instance of {@link T}.
+   *
    * @throws IOException If anything goes wrong reading from the {@code buffer}.
    */
   public <T> T read(final Class<T> type, final InputStream inputStream) throws IOException {
     Objects.requireNonNull(type);
     Objects.requireNonNull(inputStream);
 
     if (InterledgerPacket.class.isAssignableFrom(type)) {
+      //noinspection ResultOfMethodCallIgnored
       inputStream.read(); // swallow type field
     }
     return lookup(type).read(this, inputStream);
   }
 
   /**
+   * Helper method that accepts a {@link byte[]} and a type hint, and then decodes the input to the
+   * appropriate response payload.
+   *
+   * <p>NOTE: This methods wraps IOExceptions in RuntimeExceptions.
+   *
+   * @param type An instance of {@link Class} that indicates the type that should be decoded.
+   * @param data An instance of {@link byte[]} that contains bytes in a certain encoding.
+   * @param <T>  The type of object to return, based upon the supplied type of {@code type}.
+   *
+   * @return An instance of {@link T}.
+   */
+  public <T> T read(final Class<T> type, final byte[] data) {
+    Objects.requireNonNull(type);
+    Objects.requireNonNull(data);
+
+    try (ByteArrayInputStream bais = new ByteArrayInputStream(data)) {
+      if (InterledgerPacket.class.isAssignableFrom(type)) {
```

Comment:
if this check is done in read(final Class<T> type, final InputStream inputStream), do we need it here too? cant we just go 
try(ByteArrayInputStream bais ....) {
 return read(type, bais);
}
?